### PR TITLE
cnap/core: add the implementation for stream provider

### LINKED ISSUE #26: design stream provider to provide stream input

### PR DESCRIPTION
**Pull request by Hulongyin**
This PR is to fix #26 , design and implement stream provider to provide stream input.

### CODE REVIEW

File: cnap/core/stream.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+
+    def prepare(self) -> None:
+        """
+        Prepare the stream processor like verifying stream provider or registering the pipeline.
+        """
+        if not self._provider.verify():
+            LOG.error("Fail to verify the provider")
+
+
+PROVIDER_TYPES: Dict[str, type] = {
```
Comment (leyao-daily) [inline cnap/core/stream.py]: Dict[str, Type[StreamProvider]] may make the hint more clear

File: cnap/core/stream.py [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+        self._max_frame = 0
+
+    def verify(self) -> bool:
+        return True
+
+    def open(self) -> bool:
+        LOG.debug("Open file source: %s", self._file_path)
+        try:
+            self._file_path = self.file_db.get_file(self.pathname)
+        except FileNotFoundError as e:
```
Comment (leyao-daily) [inline cnap/core/stream.py:261]: I think if we catch exception, then we may use log.exception to handle

File: cnap/core/stream.py (hunk 1 above)
Comment (Hulongyin) [reply cnap/core/stream.py]: Thanks, fixed.

File: cnap/core/stream.py (hunk 2 above)
Comment (Hulongyin) [reply cnap/core/stream.py:261]: Thanks, fixed.

**Review by leyao-daily — APPROVED** [review]
LGTM.

**Comment by Hulongyin** [comment]
Closes #26

**Pull request merged**